framework/caching/FileCache.php

```diff
@@ -110,7 +110,7 @@ protected function getValue($key)
     {
         $cacheFile = $this->getCacheFile($key);
 
-        if (@filemtime($cacheFile) > time()) {
+        if (file_exists($cacheFile) && @filemtime($cacheFile) > time()) {
```

Comment:
1. It may result in additional stat call.
2. It becomes non-atomic operation.

Reply:
Yes, why do we want to avoid it at all costs?

Reply:
`@filemtime` assumes that file already exists - which is already wrong assumption. Then it solves situation with non-existing file by suppressing error - which is also [wrong approach](https://stackoverflow.com/questions/136899/suppress-error-with-operator-in-php/960288#960288). PHP has dedicated function(s) to check whether the file exists - so why not to use it properly? It is not adding extra stat call since stat data are memory cached on first attempt. The currently implemented mechanism is also not 100% atomic anyway since there is microtime between `@filemtime` and `@fopen` and `@flock` in the block (BTW would not `file_get_contents()` be better?). Should you wish I can add extra suppress operator `@file_exists($path)` even though I don't think it'd be necessary.

Reply:
@lubosdz because there's a time gap between `file_exists` and `filemtime`. `file_get_contents()` with locking flags might be, indeed, better in terms of atomicity.